Client/Utils/CumulativeDaysOfUseCounter.swift

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+// Counter to know how if a user has used the app a certain number of days in a row
+// Useful to manage the rating prompt trought RatingPromptManager
+class CumulativeDaysOfUseCounter {
+
+    private let calendar = Calendar.current
+    private let maximumNumberOfDaysToCollect = 7
+    private let requiredCumulativeDaysOfUseCount = 5
+
+    private enum UserDefaultsKey: String {
+        case keyArrayDaysOfUse = "com.moz.arrayDaysOfUse.key"
+        case keyRequiredCumulativeDaysOfUseCount = "com.moz.hasRequiredCumulativeDaysOfUseCount.key"
+    }
+
+    private(set) var hasRequiredCumulativeDaysOfUse: Bool {
+        get { UserDefaults.standard.object(forKey: UserDefaultsKey.keyRequiredCumulativeDaysOfUseCount.rawValue) as? Bool ?? false }
+        set { UserDefaults.standard.set(newValue, forKey: UserDefaultsKey.keyRequiredCumulativeDaysOfUseCount.rawValue) }
+    }
+
+    private var daysOfUse: [Date]? {
+        get { UserDefaults.standard.array(forKey: UserDefaultsKey.keyArrayDaysOfUse.rawValue) as? [Date] }
+        set { UserDefaults.standard.set(newValue, forKey: UserDefaultsKey.keyArrayDaysOfUse.rawValue) }
+    }
+
+    func updateCounter(currentDate: Date = Date()) {
+        // If there's no data, add current day of usage
+        guard var daysOfUse = daysOfUse, let lastDayOfUse = daysOfUse.last else {
+            daysOfUse = [currentDate]
+            return
+        }
+
+        // Append usage days that are not already saved
+        let numberOfDaysSinceLastUse = calendar.numberOfDaysBetween(lastDayOfUse, and: currentDate)
+        if numberOfDaysSinceLastUse >= 1 {
+            daysOfUse.append(currentDate)
+            self.daysOfUse = daysOfUse
+        }
+
+        // Check if we have 5 consecutive days in the last 7 days
+        hasRequiredCumulativeDaysOfUse = hasRequiredCumulativeDaysOfUse(daysOfUse: daysOfUse)
+
+        // Clean data older than 7 days
+        cleanDaysOfUseData(daysOfUse: daysOfUse, currentDate: currentDate)
+    }
+
+    private func hasRequiredCumulativeDaysOfUse(daysOfUse: [Date]) -> Bool {
+        var cumulativeDaysCount = 0
+        var previousDay: Date?
+        var maxNumberOfConsecutiveDays = 0
+
+        for dayOfUse in daysOfUse {
```

Comment:
Nit: When we are iterating through a whole array, I prefer using `forEach` to clarify that intent. Thus, `for in` loops communicate breakability. I am weird this way. TOTALLY NO NEED TO UPDATE.
```suggestion
        daysOfUse.forEach { dayOfUse in
```